bigchaindb_driver/connection.py

```diff
@@ -11,31 +11,42 @@
 class Connection:
     """A Connection object to make HTTP requests."""
 
-    def __init__(self, *, node_url):
+    def __init__(self, *, node_url, headers=None):
         """Initializes a :class:`~bigchaindb_driver.connection.Connection`
         instance.
 
         Args:
             node_url (str):  Url of the node to connect to.
+            headers (dict): Optional headers to send with each request.
 
         """
         self.node_url = node_url
         self.session = Session()
+        if headers:
+            self.session.headers.update(headers)
 
-    def request(self, method, *, path=None, json=None, params=None, **kwargs):
+    def request(self, method, *, path=None, json=None,
+                params=None, headers=None, **kwargs):
         """Performs an HTTP requests for the specified arguments.
 
         Args:
             method (str): HTTP method (e.g.: `'GET`').
             path (str): API endpoint path (e.g.: `'/transactions'`).
             json (dict): JSON data to send along with the request.
             params (dict)): Dictionary of URL (query) parameters.
+            headers (dict): Optional headers to pass to the request.
             kwargs: Optional keyword arguments.
 
         """
         url = self.node_url + path if path else self.node_url
         response = self.session.request(
-            method=method, url=url, json=json, params=params, **kwargs)
+            method=method,
+            url=url,
+            json=json,
+            params=params,
+            headers=headers,
```

Comment:
I assume these will override any that were added with `session.headers.update()`?